client/me/account-close/closed.jsx

```diff
@@ -1,46 +1,104 @@
-					<Spinner size={ 32 } />
-					<p className="account-close__spinner-text">
-						{ translate( 'Your account is being deleted' ) }
-					</p>
-				</div>
-			);
+	// restore token is either in the URL or in the reducer
+	const params = new URLSearchParams( window.location.search );
+	const urlToken = params.get( 'token' );
+	const storedToken = useSelector( getRestoreToken );
+	const restoreToken = urlToken || storedToken;
+
+	// Sync token to URL if not already there
+	useEffect( () => {
+		if ( storedToken && ! urlToken ) {
+			const newUrl = new URL( window.location.href );
+			newUrl.searchParams.set( 'token', storedToken );
+			window.history.replaceState( {}, '', newUrl.toString() );
 		}
+	}, [ storedToken, urlToken ] );
+
+	const onClick = () => {
+		window.location.href = '/';
+	};
 
+	const onClickRestore = () => {
+		dispatch( restoreAccount( restoreToken ) );
+	};
+
+	if ( ( ! isUserAccountClosed && ! config.isEnabled( 'me/account-restore' ) ) || ! restoreToken ) {
 		return (
-			<EmptyContent
-				title={ translate( 'Your account has been closed' ) }
-				line={ translate( 'Thanks for flying with WordPress.com' ) }
-				secondaryAction={ translate( 'Return to WordPress.com' ) }
-				secondaryActionCallback={ this.onClick }
-			/>
+			<BlankCanvas className="account-deleted">
+				<BlankCanvas.Header />
+				<BlankCanvas.Content>
+					<FormattedHeader
+						brandFont
+						headerText={ translate( 'Your account is being deleted' ) }
+						subHeaderText={ <Spinner style={ { width: '32px', height: '32px' } } /> }
+					/>
+				</BlankCanvas.Content>
+			</BlankCanvas>
 		);
 	}
+
+	return (
+		<BlankCanvas className="account-deleted">
+			<BlankCanvas.Header>
+				<Button variant="link" className="account-deleted__button-link" href="/">
+					Create an account
+				</Button>
+			</BlankCanvas.Header>
+			<BlankCanvas.Content>
+				<FormattedHeader
+					brandFont
+					headerText={ translate( 'Your account has been deleted' ) }
+					subHeaderText={
+						config.isEnabled( 'me/account-restore' )
+							? translate(
+									'Thanks for flying with WordPress.com. You have 30 days to restore your account if you change your mind.'
+							  )
+							: translate( 'Thanks for flying with WordPress.com.' )
+					}
+				/>
+				<div className="account-deleted__buttons">
+					<Button variant="secondary" onClick={ onClick }>
+						{ translate( 'Return to WordPress.com' ) }
+					</Button>
+					{ config.isEnabled( 'me/account-restore' ) && (
+						<Button
+							variant="link"
+							className="account-deleted__button-link"
+							onClick={ onClickRestore }
+							isBusy={ isRestoring }
+						>
+							{ translate( 'I made a mistake! Restore my account' ) }
+						</Button>
+					) }
+				</div>
+			</BlankCanvas.Content>
+		</BlankCanvas>
+	);
 }
 
 export default connect( ( state ) => {
 	return {
 		previousRoute: getPreviousRoute( state ),
 		isUserAccountClosed: isAccountClosed( state ),
```

Comment:
Optional, but since new `useSelector` calls have been added in this PR, you could also move `getPreviousRoute` and `isAccountClosed` into a `useSelector` hook and remove the `connect()` wrapper entirely.